src/CommunityToolkit/Xamarin.CommunityToolkit/Views/SideMenuView/SideMenuView.shared.cs

```diff
@@ -93,6 +94,29 @@ public SideMenuView()
 			#endregion
 		}
 
+		readonly WeakEventManager<SideMenuStateChangingEventArgs> stateChangingEventManager = new WeakEventManager<SideMenuStateChangingEventArgs>();
+		readonly WeakEventManager<SideMenuStateChangedEventArgs> stateChangedEventManager = new WeakEventManager<SideMenuStateChangedEventArgs>();
```

Comment:
Shouldn't we create a single weak event manager for all events like we do in other places?

```csharp
readonly WeakEventManager weakEventManager = new WeakEventManager();
```

Reply:
This is a good idea, I have added it in my latest code push